src/core/unit.h

```diff
@@ -368,6 +368,9 @@ typedef struct Unit {
         /* Most recently read value of memory.peak */
         uint64_t memory_peak_last;
 
+        /* Most recently read value of memory.swap.peak */
+        uint64_t memory_swap_peak_last;
```

Comment:
hmm, so i just noticed that both these fields lack serialization/deserialization.

See unit_serialize_state() in src/core/unit-serialize.c, where cpu_usage_last and related fields are serialized, and unit_deserialize_state() where they are deserialized

Without this the display of this data will be corrupted once you do "systemctl daemon-reload" or "systemctl daemon-reexec".

Reply:
I didn't know that this facility exists. Happy to add it. But wouldn't the unit obtain the value from `/sys/fs/cgroup` after the daemon got restarted/reloaded?

Reply:
that won't work if the unit is already gone. i mean, the reason you added the field in the first place still applies after a daemon-reload, hence it should really be serialized

Reply:
Fair point. https://github.com/systemd/systemd/pull/30008